### PR TITLE
- Add option tho show thumbs for images in content listing and rename…

### PR DESCRIPTION
**Pull request by fgrcon**
…-popover

  https://github.com/plone/Products.CMFPlone/issues/1074
  [fgrcon]

### CODE REVIEW

File: plone/app/content/browser/vocabulary.py [hunk 1]
```diff
@@ -152,6 +153,12 @@ def __call__(self):
                             continue
                     if key == 'path':
                         val = val[len(base_path):]
+                    if key == 'Thumb':
+                        rid = vocab_item.getRID()
+                        iface = 'plone.namedfile.interfaces.IImageScaleTraversable'
+                        provided = context.portal_catalog.getIndexDataForRID(rid)['object_provides']
```
Comment (vangheem) [inline plone/app/content/browser/vocabulary.py]: very clever

**Comment by vangheem** [comment]
What does the "getIcon" metadata value contain these days? I know there was talk it should be removed a while ago so I'm not even sure it's useful.

Anyone else have any comments on this? It's a bit odd having to do the extra lookup this way and it goes against the idea of the catalog vocabulary query being a straight representation of the catalog data but I'm not sure there is a better way to implement this.

**Comment by fgrcon** [comment]
@vangheem 
I did not like the approach myself (too complicated). And it does not work well with IImageScaleTraversable:
*  for now all dexterity types provide this interface see  https://github.com/plone/Products.CMFPlone/issues/1129

* also item types with leadimage would provide a thumb which is fine, but this would show up as not existing image when no leadimage provided (ugly)

-----
Pragmatic approach could be: dont look up the index at all, just return os.path.join(vocab_item.getURL(),'@@images','image','tile')  for thumbs 
then we get image tags with not existing src like:
```
       <td class="Thumb">
            <img src="http://192.168.1.15:8080/Plone/front-page/@@images/image/tile">
       </td>
```    

**Could we find a way to filter out the image tag when image src not exists?** (server side / client side ?)**
---
or: introduce a catalog metadata field " thumbcontent "  -  but this  would most probably blow up dat.fs ?

**Comment by jensens** [comment]
please note, @fgrcon needs to become core contributor before we can merge, see also plone/mockup#592

**Comment by fgrcon** [comment]
agreement submitted

**Comment by polyester** [comment]
agreement received, invitation sent

**Comment by vangheem** [comment]
@fgrcon if all types in plone already provide that interface, than I don't think we should have any vocabulary modification here and the javascript can just detect if it is a Image type(maybe we provide a configurable list of image types).

So it'd be closing this PR and updating the mockup repo a bit more. What do you think?

**Comment by fgrcon** [comment]
@vangheem  I tried /debugged quite a lot  recently ..., :
* best solution so far: new meta data field (brains) :  has_image (Booleanindex) for IDexterityContent works fine, no need  to tamper javascript  ... querying this field is better and much less resource consuming than the previous approach, will need an upgrade step though. this also covers any type with leadimages a s well
I will update pull requests soon,  after doing some more tests.

btw: plone.app contenttypes.indexers  getIcon_file seems not to be used anywhere and should maybe removed

**Comment by vangheem** [comment]
We generally advise against new metadata fields as it adds overhead. We're actually trying to reduce the number of metadata/index fields fields :(

Maybe fixing getIcon_file to work would be a better way to fix this?

It doesn't sound like a good idea to add another field regarding images/icons though.

**Comment by fgrcon** [comment]
ok i understand this. I found out that some old code (deprecated getIcon) coul be removed from plone.app.contenttypes.indexer. Now I reuse the metadafield getIcon for my purpose though this mit be confusing a little bit.Updated  Pull requests will follow.

**Comment by fgrcon** [comment]
icons are rendered with fontello icons  now. But there are quite lot locations where still geticon is in the code. E.g. in navigation portlet you get a fontello icon followed by an folder.png when you  select show icons. 
double-icons


I am trying  fo clean up and close the pr meanwhile. this leads to anumber of 404s :
WARN [web-server]: 404: /image.png
WARN [web-server]: 404: /document.png
WARN [web-server]: 404: /folder.png
see https://travis-ci.org/plone/mockup/jobs/84647351

**Comment by vangheem** [comment]
Yes, we need to fix those templates to not use that.

**Comment by fgrcon** [comment]
Just working on that

**Comment by jensens** [comment]
Whats the state here? Not needed any longer, because of the other PR? If so please close here.

**Comment by fgrcon** [comment]
@jensens  No the changeset in this pr works fine and adds a new feature,  but depends on:
* https://github.com/plone/mockup/pull/592

I wanted to introduce a new metadata field in order to implement that. @vangheem  rejected this approach and suggested to fix getIcon instead ....
Analyzing the whole code stack I found out:
* Plone 5 uses css class  "content-type" +normalized(portal_type), rendered with fontello-icon
*  We have  **much deprecated code concerning how to render icons, which renders img-tags (very often together with the fontello icon, which looks extremly ugly and increases bandwidth).** See comments above..

In order to resuse metadata field geticon (which will now bee boolean), I needed to clean up this stuff, quite a lot of work.

This will be achieved by:
* https://github.com/plone/Products.CMFPlone/pull/1152
* https://github.com/plone/plone.app.portlets/pull/59
* https://github.com/plone/plone.app.contenttypes/pull/278
* ...plone.app.contentlisting...  I am working on that right now (will fix http://jenkins.plone.org/job/pull-request-5.0/456/)
* and probably a few more commits. 
So I would suggest to wait to merge all this prs untill i am finished with cleanup (doing a lot of local tests as  well).

**Comment by jensens** [comment]
even if i'am ok to unify this all: I would like to ask the @plone/framework-team if this is something we better put into a plip and schedule for Plone 5.1? At least it breaks for some addons code.

**Comment by fgrcon** [comment]
No objection to that. That's also why I keep that stuff in my own repos.

On the other hand it might be a good idea to clean up the code stack from time to time. As far addons are concerned, the same modifications will have to be made there anyway (for Plone 5).
Nevertheless I will try to finish the work  asap.
In meantime I'll just use my private branched components.

**Comment by jensens** [comment]
yes, thats why i'd propose to schedule it for plone 5.1

**Comment by davilima6** [comment]
I'd also like to hear the framework team but since the vast majority of addons haven't been migrated to Plone 5 yet I'm not sure that would be so much of a problem.

BTW this relates to #1093.

**Comment by vangheem** [comment]
The use a getIcon right now in portlets and other things causes a rendering bug where 2 icons show in listings right now. I don't think this would require a plip.

**Comment by vangheem** [comment]
build running: http://jenkins.plone.org/job/pull-request-5.0/473

**Comment by fgrcon** [comment]
closed see: plone/Products.CMFPlone#1226

**Pull request closed without merging**